src/app/containers/ConsentBanner/index.amp.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import styled from 'styled-components';
+import Banner from './Banner/index.amp';
+
+const parentId = 'consent';
+const promptId = 'consent-prompt';
+const privacyId = 'privacy';
+const cookieId = 'cookie';
+
+const ampGeoData = {
+  ISOCountryGroups: {
+    eu: [
+      'at',
+      'be',
+      'bg',
+      'cy',
+      'cz',
+      'de',
+      'dk',
+      'ee',
+      'es',
+      'fi',
+      'fr',
+      'gb',
+      'gr',
+      'hr',
+      'hu',
+      'ie',
+      'is',
+      'it',
+      'li',
+      'lt',
+      'lu',
+      'lv',
+      'mt',
+      'nl',
+      'no',
+      'pl',
+      'pt',
+      'ro',
+      'se',
+      'si',
+      'sl',
+    ],
+  },
+};
+
+const ampConsentData = {
+  consents: {
+    'user-consent': {
+      promptIfUnknownForGeoGroup: 'eu',
+      promptUI: promptId,
+    },
+  },
+};
+
+const jsonInlinedScript = data => (
+  <script
+    type="application/json"
+    /* eslint-disable-next-line react/no-danger */
+    dangerouslySetInnerHTML={{ __html: JSON.stringify(data) }}
+  />
+);
+
+// Style `amp-consent` as child due to inability to set
+// `layout` attribute on styled `amp-consent` component
```

Comment:
I know you've now done all this work, but want to check if you've explored the styled-components option of passing custom attrs to the DOM...? https://github.com/styled-components/styled-components/issues/1250#issuecomment-377524919

Seems like this would remove the need for the workarounds you've done here. 😄

Reply:
Yeah tried loads, sadly doesnt work, they even have a test for `attrs does not pass non html tags to HTML element` https://github.com/styled-components/styled-components/blob/c3eedbeb79621acd668e15ed2d63107528ed81c3/packages/styled-components/src/test/attrs.test.js#L255-L264